comp/core/workloadmeta/collectors/internal/containerd/containerd.go

```diff
@@ -279,13 +279,28 @@ func (c *collector) notifyInitialImageEvents(ctx context.Context, namespace stri
 		return err
 	}
 
+	filteredImages := make(map[workloadmeta.EntityID]*workloadmeta.ContainerImageMetadata)
 	for _, image := range existingImages {
-		if err := c.notifyEventForImage(ctx, namespace, image, nil); err != nil {
+		wlmImage, err := c.createOrUpdateImageMetadata(ctx, namespace, image, nil)
+		if err != nil {
 			log.Warnf("error getting information for image with name %q: %s", image.Name(), err.Error())
 			continue
 		}
+		//Image is referenced by several different names but has only one entity id (= manifest.Config.Digest).
+		//filteredImages will be refreshed by updated workloadmeta.ContainerImageMetadata
+		if wlmImage != nil {
+			filteredImages[wlmImage.EntityID] = wlmImage
+		}
```

Comment:
if I understand we can have several time an image with the same EntityID. So we use a map to only keep one instance. 
But are we sure that content is the same between 2 images with the same entityID? should we merge the info?

Reply:
I think behind the hood `createOrUpdateImageMetadata` is doing a merge of the RepoTags and RepoDigests. However, there might be an issue with the image name: https://github.com/DataDog/datadog-agent/blob/f1970ac27b0b98281322a95dc53417e01ea37d91/comp/core/workloadmeta/collectors/internal/containerd/image.go#L291-L308

Reply:
references for the same image should have same name. Here is the [function](https://github.com/containerd/containerd/blob/2b661b890f689716b2928408e386e656d8dd7118/pkg/cri/server/image_pull.go#L180) that creates three references.

Reply:
Isn't it 3 different names ? The signature of the function is here:
https://github.com/containerd/containerd/blob/a2d0ddc88e1812407e6dbb829cc845175359e360/pkg/cri/server/images/image_pull.go#L313

And r is either an imageId, a repotag or a repodigest:
https://github.com/containerd/containerd/blob/2b661b890f689716b2928408e386e656d8dd7118/pkg/cri/server/image_pull.go#L176-L180

Reply:
Sorry, yes, 3 reference and 1 unique id which is image id = config.digest according to this [comment](https://github.com/containerd/containerd/blob/2b661b890f689716b2928408e386e656d8dd7118/pkg/cri/server/image_pull.go#L276)